src/lib/components/composites/PaperBookmarkButton.svelte

```diff
@@ -23,57 +28,115 @@
         }),
     );
 
+    let isUpdatingBookmarkStatus = $state(false);
+
     const onMouseEnter = () => (isHovered = true);
     const onMouseLeave = () => (isHovered = false);
 
     /**
-     * Adds the paper to the reading list, if it is not added yet, otherwise removes it.
+     * Fetches the actual bookmark status when the paper ID becomes available.
+     * Uses a cleanup function to handle cases where the paper ID changes before the fetch completes.
      */
-    function toggleBookmarkState() {
-        if (isBookmarked) {
-            removePaperFromReadingList();
+    $effect(() => {
+        const currentPaperId = paperId.value;
+        let cancelled = false;
+
+        if (typeof currentPaperId === "string") {
```

Comment:
```suggestion
        if (currentPaperId) {
```
Why do you want to check the type of the paperId.value? It is either a string or undefined and so use can use normal truthy checks?

Reply:
Yes you're right. A simple truthy check is sufficient and much cleaner!